client/extensions/woocommerce/app/products/product-form-details-card.js

```diff
@@ -29,6 +31,11 @@ export default class ProductFormDetailsCard extends Component {
 	constructor( props ) {
 		super( props );
 
+		const { product } = props;
+		this.state = {
+			updateSkuOnNameChange: ! product.name ? true : false,
```

Comment:
Why not just `updateSkuOnNameChange: ! product.name`? The logic here is that the SKU is not automatically updated when an existing product (one with a name) is being updated, right?